cirq/contrib/acquaintance/shift_swap_network.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+import itertools
+from typing import Dict, Iterable, Optional, Sequence, Tuple
+
+from cirq import ops, protocols
+from cirq.contrib.acquaintance.gates import acquaint
+from cirq.contrib.acquaintance.permutation import (
+        PermutationGate)
+from cirq.contrib.acquaintance.shift import (
+        CircularShiftGate)
+
+
+class ShiftSwapNetworkGate(PermutationGate):
```

Comment:
This seems like a specific instance of a more general concept that seems like a "nested swap network" or perhaps "tensor product of two swap networks". Is there a way for you to define a general mechanism for combining permutations over groups of qubits, instead of introducing special cases of that idea?

Reply:
I see what you're saying, but I doubt the abstraction is worth it at this point.

Reply:
Why did you add this specific instance? Did you run into it in many places?

Reply:
It's a building block for a strategy that corresponds to UCCSD.